server/data/data-jmap/src/test/java/org/apache/james/jmap/api/projections/MessageFastViewProjectionHealthCheckTest.java

```diff
@@ -0,0 +1,220 @@
+package org.apache.james.jmap.api.projections;
+
+import static org.apache.james.jmap.api.projections.MessageFastViewProjectionContract.MESSAGE_FAST_VIEW_PRECOMPUTED_PROPERTIES_1;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.stream.IntStream;
+
+import org.apache.james.core.healthcheck.ComponentName;
+import org.apache.james.core.healthcheck.Result;
+import org.apache.james.jmap.memory.projections.MemoryMessageFastViewProjection;
+import org.apache.james.mailbox.model.TestMessageId;
+import org.apache.james.metrics.tests.RecordingMetricFactory;
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import reactor.core.publisher.Mono;
+
+class MessageFastViewProjectionHealthCheckTest {
+
+    private static final ComponentName COMPONENT_NAME = new ComponentName("MessageFastViewProjection");
+
+    private static final TestMessageId MESSAGE_ID_1 = TestMessageId.of(1);
+
+    private MessageFastViewProjectionHealthCheck testee;
+    private MemoryMessageFastViewProjection fastViewProjection;
+
+    @BeforeEach
+    void setUp() {
+        RecordingMetricFactory metricFactory = new RecordingMetricFactory();
+        fastViewProjection = new MemoryMessageFastViewProjection(metricFactory);
+        testee = new MessageFastViewProjectionHealthCheck(metricFactory);
+    }
+
+    @Test
+    void componentNameShouldReturnTheRightValue() {
+        assertThat(testee.componentName())
+            .isEqualTo(COMPONENT_NAME);
+    }
+
+    @Nested
+    class WhenZeroHitCount {
+
+        @Test
+        void checkShouldReturnHealthyWhenNoRetrieveCalled() {
+            assertThat(testee.check())
+                .isEqualTo(Result.healthy(COMPONENT_NAME));
+        }
+
+        @Test
+        void checkShouldReturnDegradedWhenNoHitButSomeMiss() {
+            Mono.from(fastViewProjection.retrieve(MESSAGE_ID_1))
+                .block();
+            Mono.from(fastViewProjection.retrieve(MESSAGE_ID_1))
+                .block();
+
+            assertThat(testee.check())
+                .isEqualTo(Result.degraded(COMPONENT_NAME, "retrieveHitCount is zero but retrieveMissCount is 2"));
+        }
+    }
+
+    @Nested
+    class WhenNonZeroHitCount {
+
+        @Test
+        void checkShouldReturnHealthyWhenMissCountPercentageIsGreaterThanTenPercent() {
+            // miss
+            Mono.from(fastViewProjection.retrieve(MESSAGE_ID_1))
+                .block();
+
+            Mono.from(fastViewProjection.store(MESSAGE_ID_1, MESSAGE_FAST_VIEW_PRECOMPUTED_PROPERTIES_1))
+                .block();
+
+            // hit
+            IntStream.rangeClosed(1, 43)
```

Comment:
What is this 43 magic constant? (& how did you computed it / determined this should be right?)

Reply:
There is `TenPercent` information in the method name. Otherwise, IDK how to get/compute the right number.